### PR TITLE
feat(app): show session resource usage

### LINKED ISSUE #727: Provide information about hard drive space and usage
This issue has remnants of a general discussion about how to communicate disk usage to users. Anything that is older than February 2022 can be ignored.

Some of the issues raised here are addressed by recent PRs such as: #923.

The resource usage as well as requests should be returned in the /servers/<name> and /servers endpoints.

The response should use the following schema:
```python
class ResourceUsage(Schema):
    cpu = fields.Str(required=False)
    memory = fields.Str(required=False)
    storage = fields.Str(required=False)

class ResourceRequests(ResourceUsage):
    gpu = fields.Str(required=False)

    @pre_load
    def resolve_gpu_fieldname(self, in_data, **kwargs):
        if "nvidia.com/gpu" in in_data.keys():
            in_data["gpu"] = in_data.pop("nvidia.com/gpu")
        return in_data

class Resources(Schema):  # Simply will replace UserPodResources
    requests = fields.Nested(ResourceRequests(), required=True)
    usage = fields.Nested(ResourceRequests(), required=False)
```

The most current data on resource usage from the session can be extracted from the JupyterServer resource manifest at `.status.mainPod.resourceUsage`. Here is an example of the format of the field:

```yaml
    resourceUsage:
      cpuMillicores: 5.633821999999999
      disk:
        availableBytes: 1002876928
        totalBytes: 1006526464
        usedBytes: 3649536
      memoryBytes: 80433152
```

What is currently being returned by the notebook service is what was requested when the session was launched. With this we will return what was requested AND what is being used.

Breaking API change requires a follow up from the UI.

**NOTE**: You cannot assume that `.status.mainPod` will be there all the time and same goes for any nested fields. The JupyterServer can exist in the cluster with just `status` being blank or definitely not containing even `mainPod` when sessions are starting`. Also if the metrics server is not enabled in the k8s cluster where amalthea is running then the cpu usage will be blank. It is also possible that due to problems with the image that is running the sessions also other parameters are missing.

### PR DESCRIPTION
**Pull request by olevski**
closes #727 

NOTE to reviewers: 90% of the changes here are the result of moving files around - i.e. instead of having one large file that contains all the schemas I broke all of these up. I will annotate the PR with comments to indicate what is exactly new code. Anything that is not annotated as new code in the diff is actually old code that just got moved.

~TODO: Fix integration tests when there is agreement on the units and final format of responses with the UI.~

/deploy renku-ui=1791-servers-resources #persist

### CODE REVIEW

File: renku_notebooks/api/__init__.py
Comment (olevski) [inline renku_notebooks/api/__init__.py:1]: all of this got moved into renku_notebooks/api/health.py

File: renku_notebooks/api/schemas/autosave.py
Comment (olevski) [inline renku_notebooks/api/schemas/autosave.py:1]: Not new code - just result from splitting the single large schemas.py

File: renku_notebooks/api/schemas/cloud_storage.py
Comment (olevski) [inline renku_notebooks/api/schemas/cloud_storage.py:1]: Not new code - just result from splitting the single large schemas.py

File: renku_notebooks/api/schemas/logs.py
Comment (olevski) [inline renku_notebooks/api/schemas/logs.py:1]: Not new code - just result from splitting the single large schemas.py

File: renku_notebooks/api/schemas/servers_post.py
Comment (olevski) [inline renku_notebooks/api/schemas/servers_post.py:1]: Not new code - just result from splitting the single large schemas.py

File: renku_notebooks/api/schemas/version.py
Comment (olevski) [inline renku_notebooks/api/schemas/version.py:1]: Not new code - just result from splitting the single large schemas.py

**Comment by olevski** [comment]
@lorenzo-cavazzi this is ready now but needs a change on the ui. Because the resources response from the api doesn't just send the requests but also requests and usage.

As we discussed the units are the same as before.

**Comment by olevski** [comment]
The changes in the response are as follows:

OLD:
```
    annotations = fields.Nested(UserPodAnnotations())
    name = fields.Str()
    state = fields.Dict()
    started = fields.DateTime(format="iso", allow_none=True)
    status = fields.Nested(ServerStatus())
    url = fields.Str()
    resources = {"cpu":"1", "memory":"1G", }  <<<< Changes are only here see below
    image = fields.Str()
```

NEW:
```
    annotations = fields.Nested(UserPodAnnotations())
    name = fields.Str()
    state = fields.Dict()
    started = fields.DateTime(format="iso", allow_none=True)
    status = fields.Nested(ServerStatus())
    url = fields.Str()
    resources = { requests: {cpu:"1", memory:"1G"}, usage: {cpu:"0.25", memory:"0.50G"} }  <<< 
    image = fields.Str()
```

**Comment by olevski** [comment]
@rokroskar yes we can. I have to resolve conflicts and bring it up to date.

**Comment by olevski** [comment]
TODO: Ensure the code in the `schemas` folder is up to date with the current state of the schemas.

**Review by lorenzo-cavazzi — APPROVED** [review]
The code looks good to me! And the changes work as expected with SwissDataScienceCenter/renku-ui#1825

I believe this requires a review from a renku-notebook maintainer (i.e. @Panaetius , the only other person in the group :grin:  )

**Review by lorenzo-cavazzi — APPROVED** [review]
:rocket:

**Pull request merged**
